### ISSUE TITLE
Trucks are cutting corners and getting stuck

### ISSUE THREAD

**Issue by utdrmac** [MaFi-Games/Captain-of-Industry-issues#338]
**Describe the issue**
Trucks are skirting/cutting corners of belts causing them to get into locations that they should not be able to get.

**Expected behavior**
For trucks to drive properly and not cut corners. Even after adding barriers, trucks still cut corners.

**Additional context**
 - Game version: 0.4.3

**Attachments**
![image](https://user-images.githubusercontent.com/4413670/173491820-37d2175d-ef20-4b30-8b2c-6a6306290d33.png)

![image](https://user-images.githubusercontent.com/4413670/173492413-854d1bb0-8f35-48eb-aad0-26f80bfb76ee.png)

**Comment by vitorb-rzero** [comment]
Can confirm that it eventually happen at this spot in my map: 
![image](https://user-images.githubusercontent.com/97470022/173683731-579a43d0-d78a-48c2-a9e6-17b50c01ece5.png)

Game version: 0.4.4

**Comment by vitorboschi** [comment]
Can confirm that it eventually happen at this spot in my map:

![image](https://user-images.githubusercontent.com/71193/173686085-fe9ee8ff-20c0-49ab-9874-5236e7f5f799.png)

Game version: 0.4.4

**Comment by Envian** [comment]
I am encountering the same problem with my factory.

As a workaround, I've added barriers to "fill" the green areas I don't want trucks with red. That seems to prevent them from getting stuck, but they still hiccup on this corner.

![20220618022841_1](https://user-images.githubusercontent.com/7389255/174539427-cc25d06b-9aa7-444c-a565-3adad305405e.jpg)

Here is a quick recording of the pathfinding issues that occur around this corner: https://youtu.be/iCkmbstaxVE

[22-06-19_23-28-00_9389-autosave.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/8938061/22-06-19_23-28-00_9389-autosave.zip)

**Comment by utdrmac** [comment]
I added some walls. Trucks drive right though it. This is a 30s video. There's a section in the middle where no action takes place. Skip to 00:15:00 for most of the action.

https://user-images.githubusercontent.com/4413670/174637203-1030c2e2-e167-417b-8631-f6a2bc356994.mp4

**Comment by Erisea** [comment]
Yeah, this is constantly happening to me at various locations.  Most recent example:
![image](https://user-images.githubusercontent.com/38910557/175809291-5aa77db4-2964-4482-ae37-560058aa9900.png)

**Comment by utdrmac** [comment]
This truck isn't stuck. I just happened to catch it doing some strange off-roading over a bridge. Again, driving in the 'red' area.

![image](https://user-images.githubusercontent.com/4413670/177399103-e531702e-e393-49f5-803b-99af69cb304f.png)
